src/mixins/legal-api-mixin.ts

```diff
@@ -0,0 +1,118 @@
+// Libraries
+import axios from '@/utils/axios-auth'
+import { Component, Vue } from 'vue-property-decorator'
+import { State, Action, Getter } from 'vuex-class'
+
+// Interfaces
+import { StateModelIF, ActionBindingIF, GetterIF, IncorporationFilingIF } from '@/interfaces'
+
+/**
+ * Mixin that provides the integration with the legal api.
+ */
+@Component
+export default class LegalApiMixin extends Vue {
+  // Global state
+  @State stateModel!: StateModelIF
+
+  // Global Getters
+  @Getter isTypeBcomp!: GetterIF
+  @Getter getFilingId!: number
+  @Getter getBusinessIdentifier!: string
+
+  // Store Actions
+  @Action setNameRequestState!: ActionBindingIF
+  @Action setFilingId!: ActionBindingIF
+
+  /**
+   * Method to save a draft or real filing.
+   * @param isDraft Boolean indicating if this filing is a draft
+   * @param filingId Optional filing id if this resuming an existing draft
+   */
+  async saveFiling (filing: IncorporationFilingIF, isDraft: boolean): Promise<any> {
+    try {
+      let filingId = this.getFilingId
+
+      // If have a filing id, update an existing filing
+      if (filingId && filingId > 0) {
+        this.updateFiling(isDraft, filing, filingId)
+      } else {
+        // Set the filingId to store
+        const response = await this.createFiling(filing)
+        // Assign a filing Id from the response to the state
+        if (response && response.header) {
+          this.setFilingId(response.header.filingId)
+        } else {
+          throw new Error('invalid API response')
+        }
+      }
+    } catch (e) {
+      if (e) {
+        // TODO:  Trigger some error dialog. Will catch any errors from the Api calls
+      }
+    }
+  }
+
+  /**
+   * Method to get a filing in progress.
+   * Future: We can use this method to parse and sort the data into store.
+   * @param filingId filing id if this resuming an existing draft
+   */
+  async fetchDraft (filingId: number): Promise<any> {
+    try {
+      const filing = await this.getFiling(filingId)
+      // TODO: Parse the filing into store. Either here or using another method.
+    } catch (e) {
+      if (e) {
+        // TODO:  Trigger some error dialog. Will catch any errors from the Api calls
+      }
+    }
+  }
+
+  /**
+   * Method to make a simple axios Post request.
+   * @param isDraft Boolean indicating if this filing is a draft.
+   * @param data The object body of the request.
+   */
+  createFiling (data: object): Promise<any> {
```

Comment:
If this does not need to be called externally, I recommend setting its visibility to "private". Also for updateFiling() and getFiling() below.

Reply:
Will do, good idea, thanks!